File: src/0739-daily-temperatures/solution.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& t) {
        int N = t.size();
        vector<int> a = vector<int>(N, 0);
        stack<int> s;

        for(int i = 0; i < N; i++) {
            while(!s.empty() && t[i] > t[s.top()])
                a[s.top()] = i - s.top(), s.pop();
            s.push(i);
        }

        return a;
    }
};

File: src/0739-daily-temperatures/solution.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        result, stack = [0] * len(temperatures), []

        for i, t in enumerate(temperatures):
            while stack and t > stack[-1][0]:
                st, si = stack.pop()
                result[si] = i - si
            stack.append((t, i))

        return result

